lib/vector-core/src/ipallowlist.rs

```diff
@@ -6,7 +6,7 @@ use ipnet::IpNet;
 use vector_config::{configurable_component, Configurable, Metadata, ToValue};
 use vector_config_common::schema::{InstanceType, SchemaGenerator, SchemaObject};
 
-/// IP network allowlist settings for network components
+/// List of allowed origin IP networks.
```

Comment:
Last minute thought, I think it would be useful to expand the docs. Would it be possible to add this line to the docs:

```
IP addresses must be in CIDR notation.
```

And then could we add an example. So add this attribute:

```rust
#[configurable(metadata(docs::examples = "ip_allow_list_example()"))] 
 ```

And then this function:

```rust
const fn ip_allow_list_example() -> [&'static str; 2] { 
     ["192.168.0.0/16", "127.0.0.1/32"]
}
```

Reply:
Yeah, that makes sense. I will add it now.

Reply:
I have added 2 IPv6 examples as well, to make it clear that it is supported as well.